Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingTreeMac.mm

```diff
@@ -465,27 +465,56 @@ static bool layerEventRegionContainsPoint(CALayer *layer, CGPoint localPoint)
     pointInContentsLayer.moveBy(rootContentsLayerPosition);
 
     Vector<LayerAndPoint, 16> layersAtPoint;
-    collectDescendantLayersAtPoint(layersAtPoint, scrolledContentsLayer.get(), pointInContentsLayer, layerEventRegionContainsPoint);
+    collectDescendantLayersAtPoint(layersAtPoint, scrolledContentsLayer.get(), pointInContentsLayer);
 
     LOG_WITH_STREAM(UIHitTesting, stream << "RemoteScrollingTreeMac " << this << " scrollingNodeForPoint " << point << " (converted to layer point " << pointInContentsLayer << ") found " << layersAtPoint.size() << " layers");
 #if !LOG_DISABLED
     for (auto [layer, point] : WTF::makeReversedRange(layersAtPoint))
         LOG_WITH_STREAM(UIHitTesting, stream << " layer " << [layer description] << " scrolling node " << scrollingNodeIDForLayer(layer));
 #endif
 
-    if (layersAtPoint.size()) {
-        auto* frontmostLayer = layersAtPoint.last().first;
-        for (auto [layer, point] : WTF::makeReversedRange(layersAtPoint)) {
-            auto nodeID = scrollingNodeIDForLayer(layer);
+    Vector<LayerAndPoint, 16> reversedLayersAtPointWithEventRegion;
+    for (auto [layer, point] : WTF::makeReversedRange(layersAtPoint)) {
+        if (layerEventRegionContainsPoint(layer, point))
+            reversedLayersAtPointWithEventRegion.append({layer,point});
+    }
+
+    if (reversedLayersAtPointWithEventRegion.size()) {
+        auto* frontmostLayer = reversedLayersAtPointWithEventRegion.first().first;
+        if (!frontmostLayer)
+            return rootScrollingNode;
+
+        for (auto [layer, point]: reversedLayersAtPointWithEventRegion) {
+            auto scrollingNodeForLayer = [&] (auto layer, auto point) -> ScrollingTreeNode* {
```

Comment:
`RefPtr scrollingNodeForLayer`. Please avoid raw pointers in new code.

Reply:
The lambda should likely return a `RefPtr< ScrollingTreeNode>` too since this is what it will have internally with the changes I suggest below.